src/gui/menus/TemplatesMenu.cpp

```diff
@@ -0,0 +1,71 @@
+#include <QDir>
+
+#include "TemplatesMenu.h"
+#include "GuiApplication.h"
+#include "ConfigManager.h"
+#include "Engine.h"
+#include "embed.h"
+#include "MainWindow.h"
+#include "Song.h"
+
+TemplatesMenu::TemplatesMenu(QWidget *parent) :
+	QMenu(tr("New from template"), parent)
+{
+	connect( this, SIGNAL( aboutToShow() ), SLOT( fillTemplatesMenu() ) );
+	connect( this, SIGNAL( triggered( QAction * ) ),
+			 SLOT( createNewProjectFromTemplate( QAction * ) ) );
+}
+
+
+
+
+void TemplatesMenu::createNewProjectFromTemplate( QAction * _idx )
+{
+		if( gui->mainWindow()->mayChangeProject(true) )
+		{
+				int indexOfTemplate = this->actions().indexOf( _idx );
+				bool isFactoryTemplate = indexOfTemplate >= m_custom_templates_count;
+				QString dirBase =  isFactoryTemplate ?
+								ConfigManager::inst()->factoryTemplatesDir() :
+								ConfigManager::inst()->userTemplateDir();
+
+				const QString f = dirBase + _idx->text().replace("&&", "&") + ".mpt";
+				Engine::getSong()->createNewProjectFromTemplate(f);
+		}
+}
+
+
+
+
+
+void TemplatesMenu::fillTemplatesMenu()
+{
+		this->clear();
+
+		auto addTemplatesFromDir = [this]( QDir dir ) {
```

Comment:
This may as well become a `private` method on its own. Now that it's in its own class, it won't clutter `MainWindow`'s interface.